### PR TITLE
[Generator] Implement TypeCache.

### PR DESCRIPTION
**Pull request by dustin-wojciechowski**
This PR introduces the TypeCache class, which pulls out the public properties that existed in TypeManager for each type.

### CODE REVIEW

**Review by haritha-mohan — APPROVED** [review]

File: src/bgen/AttributeManager.cs [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 public class AttributeManager {
 	public BindingTouch BindingTouch;
-	TypeManager TypeManager { get { return BindingTouch.TypeManager; } }
+	TypeCache TypeCache { get { return BindingTouch.TypeCache; } }
```
Comment (stephen-hawley) [inline src/bgen/AttributeManager.cs:12]: minor: `TypeCache TypeCache => BindingTouch.TypeManager;`

File: src/bgen/TypeCache.cs [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+
+	public TypeCache (BindingTouch bindingTouch, Assembly apiAssembly, Assembly corlibAssembly, Assembly platformAssembly)
+	{
+		if (bindingTouch.Frameworks is null)
+			throw ErrorHelper.CreateError (3, bindingTouch.CurrentPlatform);
+		if (bindingTouch.universe is null)
+			throw ErrorHelper.CreateError (4, bindingTouch.CurrentPlatform);
+
+		/* corlib */
+		System_Attribute = Lookup (corlibAssembly, "System", "Attribute");
```
Comment (mcumming) [inline src/bgen/TypeCache.cs:145]: You may want to consider using `const string` for all the repeated strings below.

**Review by mcumming — APPROVED** [review]

**Review by rolfbjarne — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

File: src/bgen/AttributeManager.cs (hunk 1 above)
Comment (dustin-wojciechowski) [reply src/bgen/AttributeManager.cs:12]: That'd be great for all the Manager objects, I'll get these in the next PR :)

File: src/bgen/TypeCache.cs (hunk 2 above)
Comment (dustin-wojciechowski) [reply src/bgen/TypeCache.cs:145]: That's a good point, I just brought this logic over wholesale. I'll get to it in an upcoming PR!

**Pull request merged**